llama.cpp

```diff
@@ -2782,6 +2785,30 @@ static void llm_load_tensors(
                         layer.wv = ml.create_tensor(ctx, tn(LLM_TENSOR_ATTN_V,   "weight", i), {n_embd, n_embd_gqa}, backend_split);
                         layer.wo = ml.create_tensor(ctx, tn(LLM_TENSOR_ATTN_OUT, "weight", i), {n_embd, n_embd},     backend_split);
 
+                        try {
+                            layer.bq = ml.create_tensor(ctx, tn(LLM_TENSOR_ATTN_Q, "bias", i), {n_embd}, backend);
+                        } catch (const std::runtime_error& e) {
+                            if (std::string(e.what()).find("not found") != std::string::npos) layer.bq = NULL; else throw;
```

Comment:
Oof. This is a very ugly (and unreliable) way to suppress the exception. The exception message is not supposed to be part of control flow - it has placeholders that could contain arbitrary text, and could be reworded or even localized. It looks like we need a try_create_tensor that returns null when the tensor isn't found.

Reply:
`create_tensor` throw Exception "not found" with `ggml_get_tensor` to check if it exists, so it is safe here.
If check before `create_tensor` here, I have no idea how to handle ctx_meta.

I don't know if a try_create_tensor safe func would be reused, as I didn't find another use case.